src/liballoc/vec.rs

```diff
@@ -2390,6 +2390,21 @@ impl<T: Clone> From<&mut [T]> for Vec<T> {
     }
 }
 
+#[stable(feature = "vec_from_array", since = "1.42.0")]
+impl<T, const N: usize> From<[T; N]> for Vec<T>
+where
+    [T; N]: LengthAtMost32,
+{
+    #[cfg(not(test))]
+    fn from(s: [T; N]) -> Vec<T> {
+        <[T]>::into_vec(box s)
+    }
+    #[cfg(test)]
```

Comment:
Why is this gated on cfg(test)? AFAICT, it's not a lang item or so so that shouldn't matter? Could you tell us what errors you're getting without it?

Reply:
Without this I was getting strange errors in stage2 in test mode about Vec not being the same as std::vec::Vec. I took this fix from [some nearby code](https://github.com/jyn514/rust/blob/e6ad27b9ace19f47fe2c8d3108ad03b1385ec105/src/liballoc/vec.rs#L2370),  if there's a better way I'd be happy to change it.

Full error: https://github.com/rust-lang/rust/pull/68692#issuecomment-580897843

Reply:
Hm, interesting, okay. Seems suspicious but since we're doing it elsewhere also fine for now.